coffee_cmd/src/main.rs

```diff
@@ -65,9 +65,7 @@ async fn main() -> Result<(), CoffeeError> {
             };
             match coffee.upgrade(&repo, verbose).await {
                 Ok(res) => {
-                    if let Some(spinner) = spinner {
-                        spinner.finish();
-                    }
+                    spinner.map(|s| s.finish());
```

Comment:
why not `and_then`?

Reply:
wait, how did that change, seems like copilot did this job :/
I'll make the changes.